client/modules/wj-util/src/decorators.ts

```diff
@@ -0,0 +1,39 @@
+import { perfy } from "./index"
+
+/**
+ * Decorator for measuring the performance of a function or method.
+ * @example
+ * ```
+ * let perf = 0
+ * class foo {
+ *   ;@measure(time => perf = time)
+ *   method() { return "some-expensive-calculation" }
+ * }
+ * ```
+ * @param callback - Callback to fire with the performance measurement taken.
+ */
+export function measure(callback: (perf: number) => void) {
```

Comment:
This decorator is unused currently (because you can only really use decorators on classes and class methods) but it serves as an example of how to make one, so I felt it was worth keeping around.